### PR TITLE
project.py:  Specify the manifest file to use UTF-8 encoding

### PR DESCRIPTION
**Pull request by ylz0923**
Specify the manifest file to use UTF-8 encoding

### CODE REVIEW

File: src/west/app/project.py [hunk 1]
```diff
@@ -321,7 +321,7 @@ def bootstrap(self, args) -> Path:
         # Parse the manifest to get "self: path:", if it declares one.
         # Otherwise, use the URL. Ignore imports -- all we really
         # want to know is if there's a "self: path:" or not.
-        manifest = Manifest.from_data(temp_manifest.read_text(),
+        manifest = Manifest.from_data(temp_manifest.read_text('utf-8'),
```
Comment (marc-hb) [inline src/west/app/project.py:324]: [locale.getencoding()](https://docs.python.org/3/library/locale.html#locale.getencoding) is used by default. If it does not already return utf8 on a system then why should `west` ignore the system's default and hardcode it?

https://docs.python.org/3/library/functions.html#open

Please try this interactively on your system and share the output:

```
python
import locale
locale.getencoding()
```

Comment (ylz0923) [reply src/west/app/project.py:324]: >>> import locale
>>> locale.getencoding()
'cp936'
>>>

>>> import locale
>>> locale.getpreferredencoding()
'cp936'
>>> locale.getdefaultlocale()
<stdin>:1: DeprecationWarning: Use setlocale(), getencoding() and getlocale() instead
('zh_CN', 'cp936')
>>> locale.getencoding()
'cp936'
>>>

how to reproduction:
west init -m URL --mr main --mf west.yml myworkspace
  > zhe west.yml using UTF-8 encoding has Chinese annotation.
error out:
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Python311\Scripts\west.exe\__main__.py", line 7, in <module>
  File "C:\Python311\Lib\site-packages\west\app\main.py", line 1085, in main
    app.run(argv or sys.argv[1:])
  File "C:\Python311\Lib\site-packages\west\app\main.py", line 244, in run
    self.run_command(argv, early_args)
  File "C:\Python311\Lib\site-packages\west\app\main.py", line 503, in run_command
    self.run_builtin(args, unknown)
  File "C:\Python311\Lib\site-packages\west\app\main.py", line 611, in run_builtin
    self.cmd.run(args, unknown, self.topdir,
  File "C:\Python311\Lib\site-packages\west\commands.py", line 194, in run
    self.do_run(args, unknown)
  File "C:\Python311\Lib\site-packages\west\app\project.py", line 224, in do_run
    topdir = self.bootstrap(args)
             ^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\west\app\project.py", line 313, in bootstrap
    manifest = Manifest.from_data(temp_manifest.read_text(),
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\pathlib.py", line 1059, in read_text
    return f.read()
           ^^^^^^^^
UnicodeDecodeError: 'gbk' codec can't decode byte 0xaf in position 151: illegal multibyte sequence

Comment (marc-hb) [reply src/west/app/project.py:324]: Can you try `$Env:PYTHONUTF8 = 1` (in powershell) and then try west again?

Comment (ylz0923) [reply src/west/app/project.py:324]: It process successfully, use $Env:PYTHONUTF8 = 1 than west init -m URL --mr main --mf west.yml myworkspace

Comment (ylz0923) [reply src/west/app/project.py]: When other places(manifest.py) call the read_text interface, they specify the encoding method, and I think it’s necessary here as well. 😊

Comment (marc-hb) [reply src/west/app/project.py]: > When other places(manifest.py) call the read_text interface, they specify the encoding method,

Good point. Can you please fetch and test #711 instead?

**Review by marc-hb — CHANGES_REQUESTED** [review]
Superseded by larger #711

**Comment by marc-hb** [comment]
I just merged larger #711.

@ylz0923 please close this one if confirmed that #711 works for you.

Thanks for your help and patience!

**Pull request closed without merging**